programs/clearing_house/src/math/amm.rs

```diff
@@ -1283,12 +1317,87 @@ pub fn calculate_max_base_asset_amount_fillable(amm: &AMM) -> ClearingHouseResul
     )
 }
 
+pub fn calculate_net_user_pnl(amm: &AMM, oracle_price: i128) -> ClearingHouseResult<i128> {
+    let net_user_base_asset_value = amm
+        .net_base_asset_amount
```

Comment:
why don't you account for unsettled lps?

Reply:
because they are responsible for their pnl whereas the net pnl is what the amm is responsible for